// @azure/cognitiveservices-visualsearch/esm/operations/images.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/imagesMappers";
import * as Parameters from "../models/parameters";
/** Class representing a Images. */
var Images = /** @class */ (function () {
    /**
     * Create a Images.
     * @param {VisualSearchClientContext} client Reference to the service client.
     */
    function Images(client) {
        this.client = client;
    }
    Images.prototype.visualSearch = function (options, callback) {
        return this.client.sendOperationRequest({
            options: options
        }, visualSearchOperationSpec, callback);
    };
    return Images;
}());
export { Images };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var visualSearchOperationSpec = {
    httpMethod: "POST",
    path: "images/visualsearch",
    urlParameters: [
        Parameters.endpoint
    ],
    queryParameters: [
        Parameters.market,
        Parameters.safeSearch,
        Parameters.setLang
    ],
    headerParameters: [
        Parameters.xBingApisSDK,
        Parameters.acceptLanguage,
        Parameters.contentType,
        Parameters.userAgent,
        Parameters.clientId,
        Parameters.clientIp,
        Parameters.location
    ],
    formDataParameters: [
        Parameters.knowledgeRequest,
        Parameters.image
    ],
    contentType: "multipart/form-data",
    responses: {
        200: {
            bodyMapper: Mappers.ImageKnowledge
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
