// @azure/cognitiveservices-visualsearch/esm/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
export var acceptLanguage = {
    parameterPath: [
        "options",
        "acceptLanguage"
    ],
    mapper: {
        serializedName: "Accept-Language",
        type: {
            name: "String"
        }
    }
};
export var clientId = {
    parameterPath: [
        "options",
        "clientId"
    ],
    mapper: {
        serializedName: "X-MSEdge-ClientID",
        type: {
            name: "String"
        }
    }
};
export var clientIp = {
    parameterPath: [
        "options",
        "clientIp"
    ],
    mapper: {
        serializedName: "X-MSEdge-ClientIP",
        type: {
            name: "String"
        }
    }
};
export var contentType = {
    parameterPath: [
        "options",
        "contentType"
    ],
    mapper: {
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export var endpoint = {
    parameterPath: "endpoint",
    mapper: {
        required: true,
        serializedName: "Endpoint",
        defaultValue: 'https://api.cognitive.microsoft.com',
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export var image = {
    parameterPath: [
        "options",
        "image"
    ],
    mapper: {
        serializedName: "image",
        type: {
            name: "Stream"
        }
    }
};
export var knowledgeRequest = {
    parameterPath: [
        "options",
        "knowledgeRequest"
    ],
    mapper: {
        serializedName: "knowledgeRequest",
        type: {
            name: "String"
        }
    }
};
export var location = {
    parameterPath: [
        "options",
        "location"
    ],
    mapper: {
        serializedName: "X-Search-Location",
        type: {
            name: "String"
        }
    }
};
export var market = {
    parameterPath: [
        "options",
        "market"
    ],
    mapper: {
        serializedName: "mkt",
        type: {
            name: "String"
        }
    }
};
export var safeSearch = {
    parameterPath: [
        "options",
        "safeSearch"
    ],
    mapper: {
        serializedName: "safeSearch",
        type: {
            name: "String"
        }
    }
};
export var setLang = {
    parameterPath: [
        "options",
        "setLang"
    ],
    mapper: {
        serializedName: "setLang",
        type: {
            name: "String"
        }
    }
};
export var userAgent = {
    parameterPath: [
        "options",
        "userAgent"
    ],
    mapper: {
        serializedName: "User-Agent",
        type: {
            name: "String"
        }
    }
};
export var xBingApisSDK = {
    parameterPath: "xBingApisSDK",
    mapper: {
        required: true,
        isConstant: true,
        serializedName: "X-BingApis-SDK",
        defaultValue: 'true',
        type: {
            name: "String"
        }
    }
};
